jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/LambdaLineMarker.kt

```diff
@@ -82,10 +78,18 @@ class LambdaLineMarker : LineMarkerProviderDescriptor() {
     ) {
     }
 
-    private fun shouldShowLineMarker(element: PsiElement, handler: String, runtime: Runtime): Boolean =
+    private fun shouldShowLineMarker(element: PsiElement, handler: String, runtimeGroup: RuntimeGroup): Boolean =
         LambdaSettings.getInstance(element.project).showAllHandlerGutterIcons ||
-        listFunctions(element.project).any { it.handler() == handler && it.runtime() == runtime.toString() } || // Handler defined in template is valid
-        try { AwsResourceCache.getInstance(element.project).lambdaFunctions().any { it.handler == handler && it.runtime == runtime } } catch (e: LambdaException) { false } // Handler in remote Lambda is valid
+            listFunctions(element.project).any {
```

Comment:
Can we extract this and the try to helper methods to make this or statement easier to grok

Reply:
Done.